packages/patternfly-4/react-core/src/demos/PaginationTable.md

```diff
@@ -57,7 +57,135 @@ class ComplexPaginationTableDemo extends React.Component {
+          </Table>
+        )}
+        {loading && <center><Title size="3xl">Please wait while loading data</Title></center>}
+      </React.Fragment>
+    );
+  }
+}
+```
+
+## Automated pagination table demo
+
+```js
+import React from 'react';
+import { Pagination, PaginationVariant, Title } from '@patternfly/react-core';
+import { Table, TableHeader, TableBody} from '@patternfly/react-table';
+
+class ComplexPaginationTableDemo extends React.Component {
+  constructor(props) {
+    this.columns = [
+      { title: "First column" },
+      { title: "Second column" },
+      { title: "Third column" }
+    ];
+    this.defaultRows = [
+      { cells: [
+        { title: "Row 1 column 1" },
+        { title: "Row 1 column 2" },
+        { title: "Row 1 column 3" }
+      ]},
+      { cells: [
+        { title: "Row 2 column 1" },
+        { title: "Row 2 column 2" },
+        { title: "Row 2 column 3" }
+      ]},
+      { cells: [
+        { title: "Row 3 column 1" },
+        { title: "Row 3 column 2" },
+        { title: "Row 3 column 3" }
+      ]},
+      { cells: [
+        { title: "Row 4 column 1" },
+        { title: "Row 4 column 2" },
+        { title: "Row 4 column 3" }
+      ]},
+      { cells: [
+        { title: "Row 5 column 1" },
+        { title: "Row 5 column 2" },
+        { title: "Row 5 column 3" }
+      ]},
+      { cells: [
+        { title: "Row 6 column 1" },
+        { title: "Row 6 column 2" },
+        { title: "Row 6 column 3" }
+      ]},
+      { cells: [
+        { title: "Row 7 column 1" },
+        { title: "Row 7 column 2" },
+        { title: "Row 7 column 3" }
+      ]},
+      { cells: [
+        { title: "Row 8 column 1" },
+        { title: "Row 8 column 2" },
+        { title: "Row 8 column 3" }
+      ]},
+      { cells: [
+        { title: "Row 9 column 1" },
+        { title: "Row 9 column 2" },
+        { title: "Row 9 column 3" }
+      ]},
+      { cells: [
+        { title: "Row 10 column 1" },
+        { title: "Row 10 column 2" },
+        { title: "Row 10 column 3" }
+      ]},
+      { cells: [
+        { title: "Row 11 column 1" },
+        { title: "Row 11 column 2" },
+        { title: "Row 11 column 3" }
+      ]},
+      { cells: [
+        { title: "Row 12 column 1" },
+        { title: "Row 12 column 2" },
+        { title: "Row 12 column 3" }
+      ]}
+    ];
+    this.state = {
+      perPage: 10,
+      page: 1,
+      rows: this.defaultRows.slice(0, 10)
+    };
+  }
+
+  updateState(_evt, newPage, newPerPage, startIdx, endIdx) {
+    return {
+      perPage: newPerPage,
+      page: newPage,
+      rows: this.defaultRows.slice(startIdx, endIdx)
+    }
+  }
+
+  renderPagination(variant = 'top') {
+    const { page, perPage, total } = this.state;
+    return (
+      <Pagination
+        itemCount={this.defaultRows.length}
+        page={page}
+        perPage={perPage}
+        defaultToFullPage
+        onSetPage={(_evt, newPage, perPage, startIdx, endIdx) => this.setState(
```

Comment:
While nesting these function calls is nice a terse it kind of makes the example a little difficult to follow. Even though `onSetPage` and `onPerPageSelect` follow a similar path in the code, they represent very different user interactions and we could maybe benefit from being more explicit and using dedicated callbacks for each. Just a thought!